src/routes/indexable-indexation-route.php

```diff
@@ -177,17 +170,17 @@ public function __construct(
 		Indexable_Prepare_Indexation_Action $prepare_indexation_action,
 		Options_Helper $options_helper
 	) {
+		parent::__construct( $options_helper );
```

Comment:
Hmm, I would let the parent have a constructor. Especially in this case it adds some complexity.

Just add a method here: `run_indexation_action`

```php
public function run_indexation_action( Indexation_Action_Interface $indexation_action, $url ) {
    try {
        parent::run_indexation_action( $indexation_action, $url );
     }
    catch( Exception $exception) {
        $this->options_helper->set( 'indexables_indexation_reason', Indexing_Notification_Integration::REASON_INDEXING_FAILED );

       trow $exception;
    }  
}
```